src/DS/sds.c

```diff
@@ -791,7 +791,14 @@ static int ds_sds_compose_catalog_has_uri(xmlDocPtr doc, xmlNodePtr catalog, con
 	xpathCtx->node = catalog;
 
 	char *expression = oscap_sprintf("cat:uri[@uri = '%s']", uri);
+	if (expression == NULL)
+	{
+		oscap_seterr(OSCAP_EFAMILY_XML, "Error: Unable to create XPath expression.");
```

Comment:
Possible problem's origin is not `libxml` or XML-related code. I believe error's affinity should be `OSCAP_EFAMILY_OSCAP`.

Reply:
I've changed `OSCAP_EFAMILY_XML` to `OSCAP_EFAMILY_OSCAP`